services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/utils/GoalStateManager.java

```diff
@@ -113,221 +213,453 @@ public boolean shouldSkipClass(Class<?> clazz) {
-                                  .buildPartial();
-                          if (networkConfiguration.getRsType().equals(Common.ResourceType.PORT))
-                            subnetState =
-                                subnetState
-                                    .toBuilder()
-                                    .setOperationType(Common.OperationType.INFO)
-                                    .build();
-                          else
-                            subnetState =
-                                subnetState
-                                    .toBuilder()
-                                    .setOperationType(Common.OperationType.CREATE)
-                                    .build();
-
-                          subnetStateSet.add(subnetState);
-                          if (!(networkConfiguration.getOpType().equals(Common.OperationType.CREATE)
-                              && networkConfiguration
-                                  .getRsType()
-                                  .equals(Common.ResourceType.PORT))) {
-                            // lookup vpc entity
-                            final VpcEntity vpcEntity = vpcMap.get(subnetEntity1.getVpcId());
-                            Vpc.VpcConfiguration vpcConfiguration =
-                                Vpc.VpcConfiguration.newBuilder()
-                                    .setId(vpcEntity.getId())
-                                    .setCidr(vpcEntity.getCidr())
-                                    .setFormatVersion(FORMAT_REVISION_NUMBER)
-                                    .setRevisionNumber(FORMAT_REVISION_NUMBER)
-                                    .build();
-                            Vpc.VpcState vpcState =
-                                Vpc.VpcState.newBuilder()
-                                    .setConfiguration(vpcConfiguration)
-                                    .setOperationTypeValue(Common.OperationType.CREATE_VALUE)
-                                    .setOperationType(Common.OperationType.CREATE)
-                                    .build();
-                            vpcStateSet.add(vpcState);
-                          }
+                          dhcpStateList.add(dhcpState);
+                          isExistingPort = true;
+                        }
+                      }
+                      String name =
+                          portStateWithEverythingFilledNB.getName() == null
+                              ? ""
+                              : portStateWithEverythingFilledNB.getName();
+
+                      Port.PortConfiguration portConfiguration =
+                          Port.PortConfiguration.newBuilder()
+                              .setName(name)
+                              .setProjectId(portStateWithEverythingFilledNB.getProjectId())
+                              .setVpcId(
+                                  portStateWithEverythingFilledNB
+                                      .getSubnetEntities()
```

Comment:
How do we ensure that this selected SubnetEntity has the right VPC id?

Reply:
I don't understand your question, do you mean one port could belong to 2 VPC? if not then any subnet info could lead to same vpc id

Reply:
anyway I changed to use vpc entities in 7ccbc91